src/accelerate/utils/environment.py

```diff
@@ -61,17 +62,41 @@ def are_libraries_initialized(*library_names: str) -> Dict[str, bool]:
     return [lib_name for lib_name in library_names if lib_name in sys.modules]
 
 
+def get_gpu_info():
+    """
+    Gets GPU count and names using `nvidia-smi` instead of torch to not initialize CUDA.
+
+    Largely based on the `gputil` library.
+    """
+    if platform.system() == "Windows":
+        # If platform is Windows and nvidia-smi can't be found in path
+        # try from systemd rive with default installation path
+        command = spawn.find_executable("nvidia-smi")
+        if command is None:
+            command = "%s\\Program Files\\NVIDIA Corporation\\NVSMI\\nvidia-smi.exe" % os.environ["systemdrive"]
+    else:
+        command = "nvidia-smi"
+    output = subprocess.check_output([command, "-L"], universal_newlines=True)
+    lines = output.split(os.linesep)
+    gpu_count = len(lines) - 1
+    # Get name from first GPU
+    gpu_name = lines[0].split("(")[0].strip()
```

Comment:
Here we check only the first GPU, but theoretically, the user could have multiple different devices. Since we cannot know at this point which one will actually be used, shouldn't we check all of them?

Reply:
Went ahead and did so with your new version